dcmdump/src/main.rs

```diff
@@ -27,6 +29,42 @@ const ERROR_READ: i32 = -2;
 /// Exit code for when an error emerged while dumping the file.
 const ERROR_PRINT: i32 = -3;
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+enum DumpValue<T>
+where
+    T: ToString,
+{
+    TagNum(T),
+    Alias(T),
+    Num(T),
+    Str(T),
+    DateTime(T),
+    Invalid(T),
+    Nothing,
+}
+
+impl<T> fmt::Display for DumpValue<T>
+where
+    T: ToString,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let value = match self {
+            DumpValue::TagNum(v) => v.to_string().dimmed(),
+            DumpValue::Alias(v) => v.to_string().bold(),
+            DumpValue::Num(v) => v.to_string().cyan(),
+            DumpValue::Str(v) => v.to_string().yellow(),
+            DumpValue::DateTime(v) => v.to_string().green(),
+            DumpValue::Invalid(v) => v.to_string().red(),
+            DumpValue::Nothing => "(no value)".dimmed(),
```

Comment:
`dimmed` seems to not make a difference from the rest of the text here. I'd say we either try a darker shade or remove the call to `dimmed()`.

![Capture-dicom-rs-colored-dcmdump-2](https://user-images.githubusercontent.com/4738426/96422094-80c23800-11ef-11eb-9022-eb6f221e3009.PNG)

Reply:
Good to know! I switched this to `italic` since I didn't want to hard-code a darker color in case someone is using a light theme, and it still needs to return a `ColoredString`